yt_dlp/postprocessor/modify_chapters.py

```diff
@@ -34,23 +33,13 @@ def run(self, info):
         if not chapters:
             chapters = [{'start_time': 0, 'end_time': duration, 'title': info['title']}]
 
-        # TODO: Refactor _remove_marked_arrange_sponsors to not need this
-        for c in sponsor_chapters:
-            c['categories'] = [(c['category'], c['start_time'], c['end_time'])]
-
         info['chapters'], cuts = self._remove_marked_arrange_sponsors(chapters + sponsor_chapters)
         if not cuts:
             return [], info
 
-        if not info.get('__real_download'):
-            # TODO: Check using duration instead
-            if self._force_remove:
-                self.report_warning('Removing chapters multiple times may cut out unintended parts of the video')
-            else:
-                self.report_warning(
-                    f'Skipping {self.pp_key()} since the video was already downloaded. '
-                    'Use --force-remove-chapters to remove the chapters anyway')
-                return [], info
+        if not info.get('__real_download') and abs(duration - info['duration']) > 1:
+            self.report_warning(f'Skipping {self.pp_key()} since the video appears to be already cut.')
+            return [], info
```

Comment:
I think we can improve this

pseudocode
```
if real_duration != duration:
    if duration == duration_after_cut:
        to_screen(Already cut)
        return
    else if real_download:
        warn(Durations mismatch)    # Since we just downloaded it, just assume extracted duration is wrong. So proceed to cut anyway
    else:
       error(Durations mismatch.)  # Since addchapters will now break
 ```

Reply:
Implemented but without the `warn(Durations mismatch)`. Warning would be a result of extractor duration inaccuracy. Doesn't look like there is something that can be done with it, so no point to annoy the user.